docs/examples/good_practices/hpo_with_orion/README.rst

```diff
@@ -0,0 +1,333 @@
+   +
+   +# Specify max trials (here 10) to prevent a too-long run.
+   +
+   +# Then you can specify a search space for each `main.py`'s script parameter
+   +# you want to optimize. Here we optimize only the learning rate.
+   +
+   +orion hunt -n orion-example --exp-max-trials 10 python main.py --learning-rate~'loguniform(1e-5, 1.0)'
+
+.. .. literalinclude:: examples/good_practices/hpo_with_orion/job.sh
+..     :language: bash
+
+
+**main.py**
+
+.. code:: diff
+
+    # distributed/001_single_gpu/main.py -> good_practices/hpo_with_orion/main.py
+   -"""Single-GPU training example."""
+   +"""Hyperparameter optimization using Oríon."""
+   +import argparse
+    import logging
+    import os
+   +import sys
+    from pathlib import Path
+
+    import rich.logging
+    import torch
+    from torch import Tensor, nn
+    from torch.nn import functional as F
+    from torch.utils.data import DataLoader, random_split
+    from torchvision import transforms
+    from torchvision.datasets import CIFAR10
+    from torchvision.models import resnet18
+    from tqdm import tqdm
+
+   +from orion.client import report_objective
+   +
+
+    def main():
+   -    training_epochs = 10
+   -    learning_rate = 5e-4
+   -    weight_decay = 1e-4
+   -    batch_size = 128
+   +    # Add an argument parser so that we can pass hyperparameters from command line.
+   +    parser = argparse.ArgumentParser(description=__doc__)
+   +    parser.add_argument('--epochs', type=int, default=10)
+   +    parser.add_argument('--learning-rate', type=float, default=5e-4)
+   +    parser.add_argument('--weight-decay', type=float, default=1e-4)
+   +    parser.add_argument('--batch-size', type=int, default=128)
+   +    args = parser.parse_args()
+   +
+   +    training_epochs = args.epochs
+   +    learning_rate = args.learning_rate
+   +    weight_decay = args.weight_decay
+   +    batch_size = args.batch_size
+
+        # Check that the GPU is available
+        assert torch.cuda.is_available() and torch.cuda.device_count() > 0
+        device = torch.device("cuda", 0)
+
+        # Setup logging (optional, but much better than using print statements)
+        logging.basicConfig(
+            level=logging.INFO,
+            handlers=[rich.logging.RichHandler(markup=True)],  # Very pretty, uses the `rich` package.
+        )
+
+        logger = logging.getLogger(__name__)
+
+   +    logger.info(f"epochs {training_epochs}, learning rate {learning_rate}, weight decay {weight_decay}, batch size {batch_size}")
+   +
+        # Create a model and move it to the GPU.
+        model = resnet18(num_classes=10)
+        model.to(device=device)
+
+        optimizer = torch.optim.AdamW(model.parameters(), lr=learning_rate, weight_decay=weight_decay)
+
+        # Setup CIFAR10
+        num_workers = get_num_workers()
+   -    dataset_path = Path(os.environ.get("SLURM_TMPDIR", ".")) / "data"
+   +    dataset_path = Path(os.environ.get("SLURM_TMPDIR", "")) / "data"
```

Comment:
Ain't this change unnecessary and should be done instead in the base example?

Reply:
En effet, corrigé dans le dernier commit! https://github.com/mila-iqia/mila-docs/pull/198/commits/4a3b3ff1c0892ddd28f10e0e06c2e0bae290fdf0